course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala

```diff
@@ -294,11 +324,21 @@ class ContentConsumptionActor @Inject() extends BaseEnrolmentActor {
         val contentIds = request.getRequest.getOrDefault(JsonKey.CONTENT_IDS, new java.util.ArrayList[String]()).asInstanceOf[java.util.List[String]]
         val fields = request.getRequest.getOrDefault(JsonKey.FIELDS, new java.util.ArrayList[String](){{ add(JsonKey.PROGRESS) }}).asInstanceOf[java.util.List[String]]
         val contentsConsumed = getContentsConsumption(userId, courseId, contentIds, batchId, request.getRequestContext)
+        //added progress....
+        val contentsProgress = getContentsProgress(userId, courseId, contentIds, batchId, request.getRequestContext)
```

Comment:
@pritha-tarento : this function is querying the progress details table for every request. what if the caller doesn't need the progress details? it will be an unnecessary database call.. may be, we should take in a parameter in the request for fetching progress details or not.